packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -913,6 +916,52 @@ async function startWatcher(opts: SetupOpts) {
     return { finished: false }
   }
 
+  async function logError(
```

Comment:
Because this function is async, we probably need a "lock" to make sure we don't open and try to write to the same file from within multiple promises concurrently.